src/Shared/BuildTransferredException.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Build.BackEnd
+{
+    internal sealed class BuildTransferredException : Exception
+    {
+        private readonly string? _typeName;
+
+        public BuildTransferredException(
+            string? message,
+            Exception? inner,
+            string? typeName,
+            string deserializedStackTrace)
+            : base(message, inner)
+        {
+            _typeName = typeName;
+            StackTrace = deserializedStackTrace;
+        }
+
+        public override string? StackTrace { get; }
+
+        public override string ToString() => $"{_typeName ?? "Unknown"}->{base.ToString()}";
+
+        internal static Exception ReadExceptionFromTranslator(ITranslator translator)
+        {
+            BinaryReader reader = translator.Reader;
+            Exception? innerException = null;
+            if (reader.ReadBoolean())
+            {
+                innerException = ReadExceptionFromTranslator(translator);
+            }
+
+            string? message = ReadOptionalString(reader);
+            string? typeName = ReadOptionalString(reader);
+            string deserializedStackTrace = reader.ReadString();
+            BuildTransferredException exception = new(message, innerException, typeName, deserializedStackTrace)
+            {
+                Source = ReadOptionalString(reader),
+                HelpLink = ReadOptionalString(reader),
+                // HResult = reader.ReadInt32(),
```

Comment:
Hresult is sometimes important, helplink and to a lesser extent source are almost never useful.

Reply:
Keeping all for now - as this is not a hot path, so extra diagnostic info can always be handy